tests/rest/RestCloudTest.php

```diff
@@ -0,0 +1,137 @@
+  }
+
+  public function testSimple40Language(): void
+  {
+    $response = $this->http->request('GET','cloud/4.0/languages/am');
+    $this->assertStandardJsonRestResponses($response);
+    $this->assertJsonSchemaAndEquals($response, RestCloudTest::SchemaFilename40, '4.0-language-am.json');
+  }
+
+  public function test40LanguageKeyboard(): void
+  {
+    $response = $this->http->request('GET','cloud/4.0/languages/ti/gff_tir_et_7');
+    $this->assertStandardJsonRestResponses($response);
+    $this->assertJsonSchemaAndEquals($response, RestCloudTest::SchemaFilename40, '4.0-language-ti-gff_tir_et_7.json');
+  }
+
+  public function test40KeyboardLanguage(): void
+  {
+    $response = $this->http->request('GET','cloud/4.0/keyboards/gff_tir_et_7/ti');
+    $this->assertStandardJsonRestResponses($response);
+    $this->assertJsonSchemaAndEquals($response, RestCloudTest::SchemaFilename40, '4.0-keyboard-gff_tir_et_7-ti.json');
+  }
+
+  // Version 3.0
+
+  public function testSimple30Keyboard(): void
+  {
+    $response = $this->http->request('GET','cloud/3.0/keyboards/us');
+    $this->assertStandardJsonRestResponses($response);
+    $this->assertJsonSchemaAndEquals($response, RestCloudTest::SchemaFilename30, '3.0-keyboard-us.json');
+  }
+
+  public function testSimple30Language(): void
+  {
+    $response = $this->http->request('GET','cloud/3.0/languages/khm');
+    $this->assertStandardJsonRestResponses($response);
+    $this->assertJsonSchemaAndEquals($response, RestCloudTest::SchemaFilename30, '3.0-language-khm.json');
+  }
+
+  public function test30LanguageKeyboard(): void
+  {
+    $response = $this->http->request('GET','cloud/3.0/languages/en/european');
+    $this->assertStandardJsonRestResponses($response);
+    $this->assertJsonSchemaAndEquals($response, RestCloudTest::SchemaFilename30, '3.0-language-en-european.json');
+  }
+
+  public function test30KeyboardLanguage(): void
+  {
+    $response = $this->http->request('GET','cloud/3.0/keyboards/european/en');
+    $this->assertStandardJsonRestResponses($response);
+    $this->assertJsonSchemaAndEquals($response, RestCloudTest::SchemaFilename30, '3.0-keyboard-european-en.json');
+  }
+
+  // Version 2.0
+
+  public function testSimple20Keyboard(): void
+  {
+    // Note, the MySQL implementation has a bug with cloud/2.0/keyboards/ta which should return 404 but doesn't!
+    // See _invalid_2.0-keyboard-ta.json
+    // The MSSQL implementation is correct
+    $response = $this->http->request('GET','cloud/2.0/keyboards/tamil');
+    $this->assertStandardJsonRestResponses($response);
+    $this->assertJsonSchemaAndEquals($response, RestCloudTest::SchemaFilename20, '2.0-keyboard-tamil.json');
+  }
+
+  public function testSimple20Language(): void
+  {
+    $response = $this->http->request('GET','cloud/2.0/languages/ta');
+    $this->assertStandardJsonRestResponses($response);
+    $this->assertJsonSchemaAndEquals($response, RestCloudTest::SchemaFilename20, '2.0-language-ta.json');
+  }
```

Comment:
Is there a test20LanguageKeyboard() or test20KeyboardLanguage() that GETs `cloud/2.0/keyboards/european/en` ?

Reply:
There is now :)